File: leetcode-submissions/0268-missing-number/solution.java
class Solution {
    public int missingNumber(int[] nums) {
        Set<Integer> numSet = new HashSet<Integer>();
        for(int i = 0; i <= nums.length; i++) {
            numSet.add(i);
        }
        for(int num: nums) {
            numSet.remove(num);
        }
        return numSet.iterator().next();
        
    }
}

File: leetcode-submissions/0268-missing-number/solution.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        
        # O(n) Time and O(1) space
        n = len(nums)
        gaussian_sum = (n * (n + 1)) // 2
        actual_sum = sum(nums)

        return gaussian_sum - actual_sum
